features/cellular/framework/API/CellularContext.h

```diff
@@ -306,6 +306,16 @@ class CellularContext : public CellularInterface {
      */
     void cp_data_received();
 
+    virtual void do_connect_with_retry();
+
+    /** Helper method to call callback function if it is provided
+     *
+     *  @param status connection status which is parameter in callback function
+     */
+    void call_network_cb(nsapi_connection_status_t status);
+
+    virtual void do_connect() = 0;
```

Comment:
This is an API break!

Reply:
These are protected methods and already implemented in AT_CellularContext class (exepct do_connect_with_retry). This was done so that RIL and AT layers don't have copy-paste code. So is this kind of change also prohibited?

Reply:
Pure virtual will cause code changes, while a function with a default implementation will not.

Reply:
Ok, if I change this to not pure virtual but virtual it's accepted?

Reply:
Agreed fix to virtual method with default implementation

Reply:
pushed fix. Now there is a default implementation in place.